### PR TITLE
refactor: Create LocationMarkerIcon component and use it in Storybook

### PR DESCRIPTION
**Pull request by joshlarson**

### CODE REVIEW

**Review by firestack — APPROVED** [review]

File: assets/src/components/map/markers/locationMarker.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <LocationDotIcon
+      className={
+        "c-location-dot-icon" +
+        (selected ? " c-location-dot-icon--selected" : "")
+      }
+    />
+  )
+}
+
+export const LocationMarker = ({
```
Comment (lemald) [inline assets/src/components/map/markers/locationMarker.tsx:18]: **suggestion (non-blocking):** now that we also have the `<UserLocationMarker/>` component I've sort of wanted to come up with a more descriptive name for this to differentiate it (maybe something indicating that it's a geocoded location from a search result?), but just separating it out into its own module is already an improvement on its own.

Comment (joshlarson) [reply assets/src/components/map/markers/locationMarker.tsx:18]: `<LocationSearchMarker />`?

Comment (lemald) [reply assets/src/components/map/markers/locationMarker.tsx:15]: @joshlarson I like `<LocationSearchMarker/>`, were you going to make that change?

Comment (joshlarson) [reply assets/src/components/map/markers/locationMarker.tsx:15]: Thanks for reminding me though - it had completely slipped my mind.

Comment (joshlarson) [reply assets/src/components/map/markers/locationMarker.tsx]: @lemald Done!

**Review by lemald — APPROVED** [review]

**Pull request merged**